timdex_dataset_api/metadata.py

```diff
@@ -158,17 +250,16 @@ def setup_duckdb_context(self) -> DuckDBPyConnection | None:
 
         The resulting, in-memory DuckDB connection is used for all metadata queries.
         """
-        if not self.database_exists():
+        conn = duckdb.connect()
+        self.configure_duckdb_connection(conn)
+
+        if self.database_exists():
+            self._attach_database_file(conn)
+        else:
             logger.warning(
                 f"Static metadata database not found @ '{self.metadata_database_path}'. "
                 "Please recreate via TIMDEXDatasetMetadata.recreate_database_file()."
             )
```

Comment:
This is an important change as well, that will become increasingly important in future PRs.

When we initialize a `TIMDEXDatasetMetadata` instance we want to ensure there is _always_ an active and valid DuckDB context created.  This context may not have any tables or views to work with yet -- e.g. the `metadata.duckdb` doesn't exist and there are no append deltas -- but it should exist configured correctly for S3.